src/21-pubsub/index.md

```diff
@@ -0,0 +1,116 @@
+
+# this is the name of the main author(s)
+author: Jeromy Johnson
+
+# technical details required for the software, don't change these.
+baseurl: ..
+template: tmpl/layouts/post.html
+collection: posts
+---
+
+To make IPFS fast for datacenters, local area networks, and large p2p
+applications, we need a fast publish-subscribe system. [Publish-Subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern),
+called 'pubsub' for short, is a pattern often used to handle events in
+large-scale networks. 'Publishers' send messages classified by topic or content and
+'subscribers' receive only the messages they are interested in, all without direct
+connections between publishers and subscribers. This approach offers much greater
+network scalability and flexibility.
+
+We have recently merged a simple, experimental pubsub implementation into go-ipfs.
+The implementation is very far from the performance and security levels we expect
+from our long-term target. But already, this feature allows for some very useful and interesting new applications.
+
+In this post, I will explain how to get started using `ipfs pubsub` and share some
+current and future examples of applications.
+
+## Getting started with ipfs pubsub
+First, you'll need to enable the pubsub code. Make sure you're running a recent version
+of go-ipfs, or try out the pre-built floodsub binaries up on the [distributions
+page](https://dist.ipfs.io/go-ipfs/floodsub-2). Once you have that version of
+ipfs installed, start the daemon with:
+
+```sh
+> ipfs daemon --enable-pubsub-experiment
+```
+
+This will tell ipfs to create and enable the pubsub service. It also implies
+that you will only be able to use pubsub with other peers who choose to enable
+it.
+
+To subscribe to the topic `foo`, run:
+```sh
+> ipfs pubsub sub foo
+```
+
+Now, any messages for the topic `foo` will be printed to your console.
+
+To publish a message to the topic `foo`, open up another terminal and run:
+```sh
+> ipfs pubsub pub foo "hello world"
+```
+
+You should see "hello world" printed out in the first terminal.  You can also
+run the `pub` command on any other connected ipfs node and your node will
+receive the message.  Messages are routed through connected, subscribed peers.
+This means that if peers A, B, and C are all subscribed to `foo`, A is connected
+to B, and B is connected to C, but A is not directly connected to C, A will
+still receive messages that C published to `foo` through B. This can be very
+useful for routing messages in networks with poor NAT traversal or otherwise
+suboptimal connectivity.
+
+To see all the your peers with pubsub enabled, check the output of:
+```sh
+> ipfs pubsub peers
+```
+
+To see all the topics you are currently subscribed to, run:
+```sh
+> ipfs pubsub ls
+```
+
+## Applications
+With pubsub, the possibilities for distributed apps on ipfs really start to open
+up. The first thing we did was to integrate it into [Orbit](https://github.com/haadcode/orbit).
```

Comment:
Mention the work to integrate pubsub into IIIF Viewers so they can do collaborative annotation of images https://github.com/ipfs/notes/issues/240